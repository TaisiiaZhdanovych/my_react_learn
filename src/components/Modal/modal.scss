.modal {
    height: 100%;
    width: 100%;
    background-color: rgba(0, 0, 0, 0.4);
    position: fixed;
    top: 0;
    left: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    opacity: 0;
    pointer-events: none;
    transition: 0.5s;
}

.modal__container {
    display: flex;
    flex-direction: column;

    text-align: center;
    width: 516px;
    height: 270px;
    background: #e74c3c;
    border-radius: 6px;
    transform: scale(0.5);
    transition: 0.5s all;

}

.modal.active {
    opacity: 1;
    pointer-events: all;
}

.modal__container.active {
    transform: scale(1);
}

.modal__header {
    display: flex;
    justify-content: space-around;
    align-items: center;
    border-radius: 6px;
    height: 68px;
    background: #d44637
}

.modal__header-text {
    font-size: 22px;
    font-weight: 500;
    line-height: 60px;
    color: #ffffff;
}

.modal__close {
    width: 20px;
    height: 20px;
    background-color: #d44637;
    color: #ffffff;
}

.modal__main-text {
    margin-top: 39px;
    margin-left: 42px;
    margin-right: 42px;
    margin-bottom: 45px;

    font-size: 20px;
    font-weight: 400;
    font-style: normal;

    line-height: 30px;
    text-align: center;
    color: #ffffff;

}

.modal__btn-box {
    display: flex;
    gap: 10px;
    justify-content: center;

}

.modal__button {
    width: 101px;
    height: 41px;
    background: #b3382c;
    border-radius: 4px;
    border: none;
    font-size: 20px;
    color: #ffffff;
}