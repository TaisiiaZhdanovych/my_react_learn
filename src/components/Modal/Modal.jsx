
import "./modal.scss";
import { useDispatch } from "react-redux";


function Modal(props) {

  const dispatch = useDispatch()
   
  
  console.log(props);
  return (
    <div className={props.active ? "modal active" : "modal"} onClick={() => dispatch(props.onClick())}>
      <div className={props.active ? "modal__container active" : "modal__container"} onClick={(e) => e.stopPropagation()}>
        <div className="modal__header">
          <span className="modal__header-text">{props.headerText}</span>
          <span className={props.active ? "modal__close" : ""} data-testid="closeModal" onClick={() => dispatch(props.onClick())}>X</span>
        </div>
        <p className="modal__main-text">{props.mainText}</p>
        <div className="modal__btn-box">
          <button className="modal__button" onClick={() => { dispatch(props.addToCart(props.id)); dispatch(props.onClick())}}>Так</button>
       
           <button className="modal__button" onClick={() => dispatch(props.onClick())}>Ні</button>
        </div>
      </div>
    </div>
  );
}

export default Modal;






