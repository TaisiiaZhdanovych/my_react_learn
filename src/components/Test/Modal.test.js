import React from "react";
import { render } from "@testing-library/react";
import Modal from "../Modal/Modal";





jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
test("render modal", () => {
    const { asFragment } = render(<Modal/>)
    expect(asFragment(<Modal/>)).toMatchSnapshot();
})



test("modal close", () => {
    
    const { getByTestId } = render(<Modal/>);

    expect(getByTestId("closeModal")).toBeTruthy();
   
});

