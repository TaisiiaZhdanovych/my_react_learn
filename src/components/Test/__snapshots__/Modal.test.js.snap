// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render modal 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal__container"
    >
      <div
        class="modal__header"
      >
        <span
          class="modal__header-text"
        />
        <span
          class=""
          data-testid="closeModal"
        >
          X
        </span>
      </div>
      <p
        class="modal__main-text"
      />
      <div
        class="modal__btn-box"
      >
        <button
          class="modal__button"
        >
          Так
        </button>
        <button
          class="modal__button"
        >
          Ні
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
